turbovnc: 3.0.1 -> 3.0.3

anti-social opened the pull request:
###### Description of changes

Changelog: https://github.com/TurboVNC/turbovnc/blob/main/ChangeLog.md#303

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

natsukium commented on pkgs/tools/admin/turbovnc/default.nix:
```diff
@@ -17,6 +17,7 @@
 , openssl
 , pam
 , perl
+, pkgconfig
```
Is this `pkg-config`?

drupol commented on pkgs/tools/admin/turbovnc/default.nix:
```diff
@@ -28,13 +29,13 @@
 
 stdenv.mkDerivation rec {
```
I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

nh2 replied:
This is a great idea, done.

drupol commented on pkgs/tools/admin/turbovnc/default.nix:
```diff
@@ -28,13 +29,13 @@
     repo = "turbovnc";
     rev = version;
-    sha256 = "sha256-GRY6aW6Kvy5sDQRiOVz2cUgKEG0IMveh80S26/rGWdM=";
+    sha256 = "sha256-akkkbDb5ZHTG5GEEeDm1ns60GedQ+DnFXgVMZumRQHc=";
```
```suggestion
    hash = "sha256-akkkbDb5ZHTG5GEEeDm1ns60GedQ+DnFXgVMZumRQHc=";
```

drupol reviewed (changes requested):
The change LGTM.

Do you think you could make a small change and implement the `finalAttrs` pattern?

It's fine if you don't want to, but it would be a nice think.

natsukium commented:
The `turbovnc.passthru.tests` fails.
Could you check this?

drupol commented:
@anti-social Do you need some help with the feedback we provided on this PR? Do you plan to update it?

nh2 commented:
> The `turbovnc.passthru.tests` fails.

```
(EE) AIGLX error: dlopen of /var/empty/lib64/dri/swrast_dri.so failed (/var/empty/lib64/dri/swrast_dri.so: cannot open shared object file: No such file or directory)
```

From the CMake build output, this is new:

```
-- XORG_DRI_DRIVER_PATH = /var/empty/lib64/dri
```

Probably related: https://github.com/TurboVNC/turbovnc/commit/c58cdb2b7fe72685f97dee5872336f32a40fa04a

nh2 committed:
turbovnc: 3.0.1 -> 3.0.3

* The DRI related variables were changed in 3.0.2:
  https://github.com/TurboVNC/turbovnc/commit/c58cdb2b7fe72685f97dee5872336f32a40fa04a

Co-Authored-By: Alexander Koval <kovalidis@gmail.com>

nh2 commented:
@drupol I fixed the test and implemented `finalAttrs` for this.

Please take another look, and merge if it looks good. Thanks!

Also thanks @anti-social!
